cmd/metrics-v3-system-drive.go

```diff
@@ -68,15 +90,147 @@ var (
 	driveAPILatencyMD = NewGaugeMD(driveAPILatencyMicros,
 		"Average last minute latency in µs for drive API storage operations",
 		append(allDriveLabels, apiL)...)
+	driveHealingMD = NewGaugeMD(driveHealing,
+		"Is it healing?", allDriveLabels...)
+	driveOnlineMD = NewGaugeMD(driveOnline,
+		"Is it online?", allDriveLabels...)
 
 	driveOfflineCountMD = NewGaugeMD(driveOfflineCount,
 		"Count of offline drives")
 	driveOnlineCountMD = NewGaugeMD(driveOnlineCount,
 		"Count of online drives")
 	driveCountMD = NewGaugeMD(driveCount,
 		"Count of all drives")
+
+	// iostat related
+	driveReadsPerSecMD = NewGaugeMD(driveReadsPerSec,
+		"Reads per second on a drive",
+		allDriveLabels...)
+	driveReadsKBPerSecMD = NewGaugeMD(driveReadsKBPerSec,
+		"Kilobytes read per second on a drive",
+		allDriveLabels...)
+	driveReadsAwaitMD = NewGaugeMD(driveReadsAwait,
+		"Average time for read requests served on a drive",
+		allDriveLabels...)
+	driveWritesPerSecMD = NewGaugeMD(driveWritesPerSec,
+		"Writes per second on a drive",
+		allDriveLabels...)
+	driveWritesKBPerSecMD = NewGaugeMD(driveWritesKBPerSec,
+		"Kilobytes written per second on a drive",
+		allDriveLabels...)
+	driveWritesAwaitMD = NewGaugeMD(driveWritesAwait,
+		"Average time for write requests served on a drive",
+		allDriveLabels...)
+	drivePercUtilMD = NewGaugeMD(drivePercUtil,
+		"Percentage of time the disk was busy",
+		allDriveLabels...)
 )
 
+func getCurrentDiskMetrics() map[string]madmin.DiskIOStats {
+	var types madmin.MetricType = madmin.MetricsDisk
+	driveRealtimeMetrics := collectLocalMetrics(types, collectMetricsOpts{
+		hosts: map[string]struct{}{
+			globalLocalNodeName: {},
+		},
+	})
+
+	stats := map[string]madmin.DiskIOStats{}
+	for d, m := range driveRealtimeMetrics.ByDisk {
+		stats[d] = m.IOStats
+	}
+	return stats
+}
+
+func setDriveBasicMetrics(m MetricValues, drive madmin.Disk, labels []string) {
```

Comment:
Pass a pointer, since you are writing to m.
```suggestion
func setDriveBasicMetrics(m *MetricValues, drive madmin.Disk, labels []string) {
```

.. actually you should do...

```suggestion
func (m *MetricValues) setDriveBasicMetrics(drive madmin.Disk, labels []string) {
```

Reply:
thanks, done